server.js

```diff
@@ -0,0 +1,341 @@
+    <!DOCTYPE html>
+    <html lang="en">
+    
+    <head>
+      <meta charset="UTF-8">
+      <title>The Elements - ${bodyData.elementName}</title>
+      <link rel="stylesheet" href="/css/styles.css">
+    </head>
+    
+    <body>
+      <h1>${bodyData.elementName}</h1>
+      <h2>${bodyData.elementSymbol}</h2>
+      <h3>Atomic number ${bodyData.elementAtomicNumber}</h3>
+      <p>${bodyData.elementDescription}</p>
+      <p><a href="/">back</a></p>
+    </body>
+    
+    </html>
+    `;
+
+            fs.writeFile(`./public/${bodyData.elementName.toLowerCase()}.html`, newPageTemplate, (err, data) => {
+              if (err) {
+                res.writeHead(500);
+                return res.end('{ "success": false }');
+              }
+              fs.readdir('./public', (err, files) => {
+                if (err) { throw new Error(err); }
+                else {
+                  let blacklist = ['404.html', 'index.html', '.keep', 'css'];
+                  let htmlList = "";
+                  let elementCount = 0;
+
+                  files.forEach(file => {
+                    if (!blacklist.includes(file)) {
+                      elementCount++;
+                      let fileName = file.slice(0, file.indexOf('.'));
+                      fileName = fileName[0].toUpperCase() + fileName.slice(1);
+
+                      htmlList += `
+                        <li>
+                          <a href="${file}">${fileName}</a>
+                        </li>
+                      `;
+                    }
+                  });
+
+                  let newIndexTemplate = `
+    <!DOCTYPE html>
+    <html lang="en">
+    
+    <head>
+      <meta charset="UTF-8">
+      <title>The Elements</title>
+      <link rel="stylesheet" href="/css/styles.css">
+    </head>
+    
+    <body>
+      <h1>The Elements</h1>
+      <h2>These are all the known elements.</h2>
+      <h3>These are ${elementCount}</h3>
+      <ol>
+        ${htmlList}
+      </ol>
+    </body>
+    
+    </html>
+    `;
+
+                  fs.writeFile('./public/index.html', newIndexTemplate, (err, data) => {
+                    if (err) {
+                      console.log(error);
+                    }
+                  });
+                }
+              });
+              res.writeHead(200, { 'Content-Type': 'application/json' })
+              return res.end('{ "success": true }');
+            });
+          }
+        });
+      } else {
+        res.writeHead(401, { "WWW-Authenticate": 'Basic realm="SecureArea"' });
+        return res.end(`<html><body>Invalid Authentication Credentials</body></html>`);
+      }
+    } else {
+      res.writeHeader(401, { 'WWW-Authenticate': 'Basic realm="SecureArea"' });
+      return res.end(`<html><body>Not Authorized</body></html>`);
+    }
+
+    /*
+    * PUT METHOD
+    */
+
+  } else if (req.method === "PUT") {
+    if (req.headers.authorization) {
+      const encodedString = req.headers.authorization.slice(6);
```

Comment:
This entire authorization section could be pulled out into it's own function since it's being used multiple times.